fe/fe-core/src/main/java/org/apache/doris/cloud/backup/CloudRestoreJob.java

```diff
@@ -0,0 +1,649 @@
+        }
+    }
+
+    private AutoCloseConnectContext buildConnectContext() throws UserException {
+        if (Strings.isNullOrEmpty(cloudClusterName)) {
+            throw new UserException("cloud cluster name is not set.");
+        }
+        if (ConnectContext.get() == null) {
+            ConnectContext ctx = new ConnectContext();
+            ctx.setCloudCluster(cloudClusterName);
+            return new AutoCloseConnectContext(ctx);
+        } else {
+            ConnectContext.get().setCloudCluster(cloudClusterName);
+            return null;
+        }
+    }
+
+    @Override
+    public synchronized void run() {
+        if (state == RestoreJobState.FINISHED || state == RestoreJobState.CANCELLED) {
+            return;
+        }
+        try (AutoCloseConnectContext r = buildConnectContext()) {
+            super.run();
+        } catch (UserException e) {
+            LOG.error("failed to run cloud restore job", e);
+        }
+    }
+
+    @Override
+    public void checkIfNeedCancel() {
+        super.checkIfNeedCancel();
+        if ((cloudClusterId = ((CloudSystemInfoService) Env.getCurrentSystemInfo()).getCloudClusterIdByName(
+                cloudClusterName)) == null) {
+            status = new Status(Status.ErrCode.NOT_FOUND, "cluster " + cloudClusterName
+                    + " has been removed");
+        }
+    }
+
+    @Override
+    public void checkStorageVault(OlapTable localTable) {
+        Preconditions.checkNotNull(storageVaultName);
+        if (((CloudEnv) Env.getCurrentEnv()).getEnableStorageVault()) {
+            if (localTable.getStorageVaultId().isEmpty()) {
+                status = new Status(Status.ErrCode.COMMON_ERROR, "local table " + localTable.getName()
+                        + " has no storage vault.");
+                return;
+            }
+            String localStorageVaultName = localTable.getStorageVaultName();
+            if (!localStorageVaultName.equals(storageVaultName)) {
+                // currently we only support unique storage vault name in one restore job
+                status = new Status(Status.ErrCode.COMMON_ERROR,
+                        "local table " + localTable.getName() + " storage vault is " + localStorageVaultName
+                                + ", but restore job storage vault is " + storageVaultName);
+            }
+        }
+    }
+
+    @Override
+    public void doCreateReplicas() {
+        try {
+            handleMetaObject(MetaSeriviceOperation.PREPARE);
+            // send create tablets requests
+            boolean needSetStorageVault = ((CloudEnv) Env.getCurrentEnv()).getEnableStorageVault();
+            for (Map.Entry<OlapTable, Cloud.CreateTabletsRequest.Builder> entry : tabletsPerTable.entrySet()) {
+                OlapTable table = entry.getKey();
+                Cloud.CreateTabletsRequest.Builder requestBuilder = entry.getValue();
+                Cloud.CreateTabletsResponse resp = sendCreateTabletsRequests(requestBuilder, table,
```

Comment:
What if there are many tablets in a table.

Reply:
We will separate create tablet task in future.